pkg/operator/ceph/disruption/clusterdisruption/add.go

```diff
@@ -36,6 +37,69 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
+// Watch for CephBlockPools and enqueue the CephCluster in the namespace
+// Watch for CephFileSystems and enqueue the CephCluster in the namespace
+// Watch for CephObjectStores and enqueue the CephCluster in the namespace
+var objectsToWatch = []client.Object{
+	&cephv1.CephBlockPool{},
+	&cephv1.CephFilesystem{},
+	&cephv1.CephObjectStore{},
+}
+
+func cephClusterPredicate[T client.Object]() predicate.TypedFuncs[T] {
```

Comment:
I see this called with `cephClusterPredicate[*cephv1.CephCluster]()` below. Is there a reason not to make this a CephCluster type here in the definition like `pdbPredicate` below?